docs/experiments/taint-propagators.md

```diff
@@ -49,14 +49,13 @@ pattern-propagators:
 ```
 
 In the example above, whenever Semgrep finds the pattern `$X.add($Y)`, it checks if
-`$Y` is tainted. If `$Y` is tainted, Semgrep propagates that taint to `$X`. Thus,
-adding tainted data to a set marks the set itself as tainted.
+`$Y` is tainted. If `$Y` is tainted, Semgrep propagates that taint to `$X`. 
+Thus, adding tainted data to a set marks the set itself as tainted.
 
 <iframe src="https://semgrep.dev/embed/editor?snippet=7lNe" border="0" frameBorder="0" width="100%" height="432"></iframe>
 
-## Propagator rule requirements
+## Requirements of taint propagator rules
 
-A propagator rule necessitates:
+A taint propagator rules necessitate:
 - Any arbitrary pattern that must bind two metavariables of your choice.
```

Comment:
what is this referring to?

Reply:
@efortuna looking at our example rule, we have - `- pattern: $X.add($Y)` - what do you think about the rewrite below?

```suggestion
- Rule pattern that binds two metavariables. For example: `- pattern: $X.add($Y)`
```

Reply:
Adam your suggestion doesn't fit with the line above "A taint propagator rule necessitates (or must contain)"

but my question was more like, what are these two metavariables? What is their function? why are we binding them? it's really vague and I don't understand what it's talking about.

Reply:
@efortuna you are right. I cannot guess this information. @IagoAbal please, can you help us out here? (See the comment above.)

Reply:
```suggestion
- Any arbitrary pattern that must bind two metavariables of your choice. For example `- pattern: $X.add($Y)`. Then specify which falls under `from` and which falls under `to`.
```

Reply:
Y'all, my point in my whole line of questioning was pointing out we literally don't say what to do with this arbitrary pattern or what it's for (and "arbitrary" is odd here. All patterns specified in a semgrep rule are "arbitrary"...). As written, this arbitrary pattern is just floating in space. The point is it is connected to the `pattern` key, and this pattern indicates where taint is propagated!

```suggestion
- A `pattern` containing two metavariables, describing where taint is propagated _to_ and _from_.
```

Reply:
```suggestion
- A `pattern` containing **two** metavariables. These two metavariables specify where taint is propagated **from** and **to**. For example, pattern `$X.add($Y)` includes two metavariables `$X` and `$Y`. Then specify which metavariable falls under `from` and which falls under `to` (see more information below).
```